docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,876 @@
+
+Last login: Fri Jan 26 21:49:12 UTC 2024 from 169.254.143.2 on pts/1
+admin@sonic:~$ 
+admin@sonic:~$ 
+Terminating...
+Thanks for using picocom
+root@sonic:/home/cisco#
+
+```
+#### Pass/Fail Criteria
+ * Verify Login access is displayed.
+ * cntrl+a and then cntrl+x to come out of the dpu console.
+ 
+ 
+### 1.5 Check midplane ip address between NPU and DPU 
+
+#### Steps
+ * Use command `show ip interface` to get ip addresses 
+ * Get the number of dpu modules from PMON APIs - get_num_modules()
+ * Currently all the interfaces gets static ip address
+ * Use the command `lspci -d 1dd8:1004` to list all the pcie buses for DPUs
+ * Ip adddress mapping to dpu - 169.254.x.2 (switch side interface) and 169.254.x.1 (DPU side interface). where x - bus number in decimal number.
+ * Work in progress - Dymanic assignment of ip address via dhcp
+
+#### Verify in
+ * Switch
+
+#### Sample Output
+```
+    On Switch:
+
+    root@sonic:/home/cisco# lspci -d 1dd8:1004
+18:00.0 Ethernet controller: Pensando Systems DSC Management Controller
+1c:00.0 Ethernet controller: Pensando Systems DSC Management Controller
+20:00.0 Ethernet controller: Pensando Systems DSC Management Controller
+24:00.0 Ethernet controller: Pensando Systems DSC Management Controller
+8b:00.0 Ethernet controller: Pensando Systems DSC Management Controller
+8f:00.0 Ethernet controller: Pensando Systems DSC Management Controller
+93:00.0 Ethernet controller: Pensando Systems DSC Management Controller
+97:00.0 Ethernet controller: Pensando Systems DSC Management Controller
+root@sonic:/home/cisco# 
+
+
+      root@sonic:/home/cisco# show ip interface
+      Interface     Master    IPv4 address/mask    Admin/Oper    BGP Neighbor    Neighbor IP
+      ------------  --------  -------------------  ------------  --------------  -------------
+      eth0                    172.25.42.65/24      up/up         N/A             N/A
+      eth1                    169.254.24.2/24      up/up         N/A             N/A
+      eth2                    169.254.28.2/24      up/up         N/A             N/A
+      eth3                    169.254.32.2/24      up/up         N/A             N/A
+      eth4                    169.254.36.2/24      up/up         N/A             N/A
+      eth5                    169.254.139.2/24     up/up         N/A             N/A
+      eth6                    169.254.143.2/24     up/up         N/A             N/A
+      eth7                    169.254.147.2/24     up/up         N/A             N/A
+      eth8                    169.254.151.2/24     up/up         N/A             N/A
+      lo                      127.0.0.1/16         up/up         N/A             N/A
+      root@sonic:/home/cisco# 
+```
+#### Pass/Fail Criteria
+ * Verify output on switch to see all 169.254.x.x network interfaces are showing both Admin and Oper up.
+ * Verify number of interfaces should be equal to number of dpu modules. 
+   
+   
+### 1.6 Check DPU shutdown and power up individually
```

Comment:
The root/shutdown flows are part of https://github.com/sonic-net/SONiC/pull/1699 HLD. I think the reboot and shutdown tests should be defined in that document

Reply:
Looks like there is an overlap, the reboot is part of the document specified above. Here as a platform testing, the test of power on/off is done through this.

Reply:
The HDL introduces new functionality and covers both reboot and power on/off flows. The assumptions here contradict the HDL. I'm ok to have those tests here but they should be aligned accordingly

Reply:
This test case is about power off and power on. It is just the cli's name is shutdown for power off case.
This is not concerned with reboot specified in that HLD document https://github.com/sonic-net/SONiC/pull/1699.

Reply:
I remember Ramesh mentioning to follow https://github.com/sonic-net/SONiC/pull/1699 HLD for shutdown and power-up sequence. @nissampa could you check with Ramesh on this?

Reply:
@vvolam @nissampa is this addressed?

Reply:
Yes, we discussed this in a review meeting and decided to keep this case as is.